### PR TITLE
fix: update for latest SDK changes

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java [hunk 1]
```diff
@@ -460,15 +460,19 @@ private QuarkusControllerConfiguration createControllerConfiguration(
...
+                        cfg = new QuarkusKubernetesDependentResourceConfig(dependentNamespaces, labelSelector);
                     }
 
                     var nameField = dependentConfig.value("name");
                     final var name = Optional.ofNullable(nameField)
                             .map(AnnotationValue::asString)
                             .filter(Predicate.not(String::isBlank))
                             .orElse(DependentResource.defaultNameFor(dependentClass));
+                    final var spec = dependentResources.get(name);
+                    if (spec != null) {
```
Comment (Sgitario) [inline core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java:472]: Why this change is needed?

File: core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java [hunk 2]
```diff
@@ -398,7 +398,7 @@ private QuarkusControllerConfiguration createControllerConfiguration(
                     getLabelSelector(controllerAnnotation),
                     Optional.ofNullable(specClassName),
                     Optional.ofNullable(statusClassName),
-                    dependentResources);
+                    dependentResources.values().stream().collect(Collectors.toUnmodifiableList()));
```
Comment (Sgitario) [inline core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java:401]: Why not use directly a list instead of creating a map in the `createDependentResources` method?

File: core/runtime/src/main/java/io/quarkiverse/operatorsdk/runtime/QuarkusKubernetesDependentResourceConfig.java [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 public class QuarkusKubernetesDependentResourceConfig extends KubernetesDependentResourceConfig {
 
     @RecordableConstructor
-    public QuarkusKubernetesDependentResourceConfig(String[] namespaces, String labelSelector) {
+    public QuarkusKubernetesDependentResourceConfig(Set<String> namespaces, String labelSelector) {
         super(namespaces, labelSelector);
     }
 
-    public String[] getNamespaces() {
+    public Set<String> getNamespaces() {
```
Comment (Sgitario) [inline core/runtime/src/main/java/io/quarkiverse/operatorsdk/runtime/QuarkusKubernetesDependentResourceConfig.java:15]: I've just found that it was not working for me because I deleted these two getters. Can you add a comment saying that the getters are necessary for the recordable constructor?

Comment (metacosm) [reply core/runtime/src/main/java/io/quarkiverse/operatorsdk/runtime/QuarkusKubernetesDependentResourceConfig.java:15]: Yes, I have similar comments on other classes, I wish Quarkus would make this explicit… maybe with an annotation.

File: core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java (hunk 2 above)
Comment (metacosm) [reply core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java:401]: So that it's easier to check if a dependent resource with the same name is already configured.

File: core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java (hunk 1 above)
Comment (metacosm) [reply core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java:472]: It wasn't needed but it's similar to what the SDK does: it shouldn't be possible to add dependent resources with the same name several times.
